ldap/servers/slapd/back-ldbm/findentry.c

```diff
@@ -32,8 +32,12 @@ check_entry_for_referral(Slapi_PBlock *pb, Slapi_Entry *entry, char *matched, co
     struct berval **url = NULL;
 
     /* if the entry is a referral send the referral */
+    if (!slapi_entry_flag_is_set(entry, SLAPI_ENTRY_FLAG_REFERRAL)) {
+        return 0;
+    }
+
     if (slapi_entry_attr_find(entry, "ref", &attr)) {
-        // ref attribute not found
+        // ref attribute not found (should not happen)
```

Comment:
If this shouldn't happen, can we add a DEBUG_ASSERT here?

Reply:
> If this shouldn't happen, can we add a DEBUG_ASSERT here?

Fixed

Reply:
@Firstyear - I made your requested change, can you approve it?

Reply:
I like ASSERT but it hides the benefit of the fix.

The issue is that most of the time the entry has no "ref" attribute. slapi_entry_attr_find("ref) will go through the complete list of attributes, doing strcasecmp on each of them to finally find nothing and return 0. Avoiding slapi_entry_attr_find to test referral is the purpose of that fix. Can we put the test in #ifdef DEBUG ?

Reply:
Sorry after a second thought I realize my mistake. slapi_entry_attr_find is only called when the entry is a referral, that is quite rare. No problem in such case to call slapi_entry_attr_find